packages/background/src/utils/notifications.ts

```diff
@@ -1,217 +1,267 @@
-    if (!chainId) {
-        return;
-    }
-
-    const explorerUrl = getExplorerUrl(chainId);
-    if (!explorerUrl) {
-        return;
-    }
-
-    addOnClickListener();
-
-    const { hash } = transactionParams;
-
-    const url = createCustomExplorerLink(hash as string, explorerUrl);
-    const title = 'Blank interaction failed';
-    const message = 'Privacy Smart Contract interaction failed!';
+/**
+ * Gets the network data from the chain list.
+ * @param chainId - The chain id.
+ * @returns The network data.
+ *
+ */
+const getNetworkData = (chainId: number | undefined) => {
+    if (!chainId || isNaN(chainId)) return undefined;
 
-    showNotification(title, message, url);
-};
+    const networkData = getChainListItem(chainId);
 
-const showRejectedTransaction = (message: string) => {
-    addOnClickListener();
+    if (!networkData || !networkData.explorers || !networkData.explorers.length)
+        return undefined;
 
-    const title = 'Transaction was rejected';
+    const explorerUrl = networkData.explorers.find((e) => e.url)?.url;
+    if (!explorerUrl) return undefined;
 
-    showNotification(title, message, '');
+    return { ...networkData, explorerUrl } as ChainListItemWithExplorerUrl;
 };
 
-const showRejectedBlankInteraction = (message: string) => {
-    addOnClickListener();
-
-    const title = 'Blank interaction rejected';
-
-    showNotification(title, message, '');
-};
+const getTxNotificationData = (
+    txMeta: TransactionMeta,
+    network: ChainListItemWithExplorerUrl
+) => {
+    const {
+        transactionParams: txParams,
+        transactionCategory,
+        status,
+        exchangeParams,
+        bridgeParams,
+        approveAllowanceParams,
+        transferType,
+    } = txMeta;
+
+    const {
+        name: txNetworkName,
+        nativeCurrency: txNetworkNativeToken,
+        explorerUrl,
+    } = network;
 
-const showNotification = (title: string, message: string, url: string) => {
-    chrome.notifications.create(url, {
-        title: title,
-        message: message,
-        iconUrl: chrome.runtime.getURL('icons/icon-48.png'),
-        type: 'basic',
-    });
-};
+    let title = '';
+    let message = '';
+    let url = '';
 
-const addOnClickListener = () => {
-    if (!chrome.notifications.onClicked.hasListener(linkToEtherscan)) {
-        chrome.notifications.onClicked.addListener(linkToEtherscan);
+    if (txParams.hash) {
+        url = createCustomExplorerLink(txParams.hash, explorerUrl);
     }
-};
 
-const linkToEtherscan = (url: string) => {
-    if (url.startsWith('https://')) {
-        chrome.tabs.create({ url: url });
+    const isTxFailed = status === TransactionStatus.FAILED;
+    const isTxCancelled = status === TransactionStatus.CANCELLED;
+
+    // Used when the transaction category is not supported or there is no data for the transaction category.
+    let showDefaultNotification = false;
+
+    switch (transactionCategory) {
```

Comment:
the only improvement I can imagine here is having a collection of notification content by type and accessing it using the category as an index, then replacing the parameters with actual data. Typing could be something like this : 
```typescript
type Notification = {
    title: string,
    message: string,
    url: string
}

type NotificationsContent = {
    [key: string]: Notification;
};


const NotificationContent: NotificationsContent = {
    'exchange': {
        title: "Swap",
        message: "${fromToken} to ${toToken} swap on ${network}",
        url: ""
    },
}    
``` 

but even without this, logic is quite clear and concise 👏🏻